### PR TITLE
Update sed to version 4.4

### PR DESCRIPTION
**Pull request by nsoranzo**
Also:
- Re-render with conda-smithy
- Add `gettext` requirement for NLS support
- Skip broken test under Linux

### CODE REVIEW

**Comment by nsoranzo** [comment]
I think I indeed had messed up my CircleCI settings, fixed  them now (and PR is green!).

**Comment by nsoranzo** [comment]
Ping @jakirkham

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
 
 build:
   number: 0
   skip: true  # [win]
 
+requirements:
+  build:
+    - gettext >=0.19.2
+  run:
+    - gettext  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why is there an OS X selector here if it is used on Linux too?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,22 +1,29 @@
...
+  url: http://ftp.gnu.org/gnu/sed/sed-{{ version }}.tar.xz
   sha256: {{ sha256 }}
 
 build:
   number: 0
   skip: true  # [win]
 
+requirements:
+  build:
+    - gettext >=0.19.2
```
Comment (jakirkham) [inline recipe/meta.yaml]: Why this version? Also why is this not reflected in run below?

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
+  run:
+    - gettext  # [osx]
+
 test:
   commands:
     - test -f "${PREFIX}/bin/sed"  # [unix]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Not your doing, but we can probably lose this selector.

**Comment by jakirkham** [comment]
Can you also please take a look, @tfors?

File: recipe/meta.yaml (hunk 1 above)
Comment (nsoranzo) [reply recipe/meta.yaml]: `gettext` is needed at runtime only on OSX because of the rpath trick used in `build.sh` .

File: recipe/meta.yaml (hunk 3 above)
Comment (nsoranzo) [reply recipe/meta.yaml]: Fixed.

File: recipe/meta.yaml (hunk 2 above)
Comment (nsoranzo) [reply recipe/meta.yaml]: It's a build requirement from `sed` itself, see https://git.savannah.gnu.org/cgit/sed.git/tree/bootstrap.conf#n95

File: recipe/meta.yaml (hunk 1 above)
Comment (jakirkham) [reply recipe/meta.yaml]: Still not following. Aren't we building with `gettext` on Linux? Why are we not requiring this on Linux? Is there some reason this isn't linked to on Linux?

Comment (nsoranzo) [reply recipe/meta.yaml]: You're right, under Linux `sed` uses the `gettext` function included in `libc`, I've added an `[osx]` selector also for the build requirement.

**Comment by nsoranzo** [comment]
Ping @jakirkham.

**Comment by jakirkham** [comment]
Might be worth adding yourself here if this package is useful to you @nsoranzo.

**Comment by jakirkham** [comment]
Thanks @nsoranzo.

**Pull request merged**